### DISCUSSION TITLE
Horrible syntax changes based on popularity or poll

### DISCUSSION THREAD

**Discussion by Wajinn** [vlang/v#18771] [General]
Because a kind of syntax appears popular or trendy in other languages, arguably doesn't mean it's necessary to make divisive changes that alienate many of the language's original fans.

If a poll is conducted, not everyone is there to support the language, but can be there to cheer on destructive changes.  Hopefully changes continue to be guided by fundamental core philosophies about the language, versus allowing something arbitrary or disruptive changes based on whim or fad.  Change can be good, but clearly care must be taken, to do what's best.

Glad that the better syntax won out (for[ static type methods](https://github.com/vlang/v/blob/master/doc/docs.md#static-type-methods)) on the twitter poll, but it was too close for comfort and clearly there are those that disagree with the outcome.  The concern is about the next time.  It's never too late for a language to do a lot of harm to itself.

"You already have things like `User.fields` for compile-time reflection. No reason to invent something new. Keep it consistent. Continue on with `.` syntax. :-) "--Rasmus Schultz @mindplaydk

"Simplicity is the key win for V. The same has contributed to steep learning curve for a lot of programmers. Introducing more syntactic sugar such as :: will do no good. In case of static I would prefer a `.`" --N PAVAN KUMAR RAO @navulepavan (author of Getting Started with V programming).

Unfortunately, many may not be able to see the original V poll (and above quotes) on Twitter as Elon appears to have wrecked Twitter for the time being.

**Comment by tzSharing** [comment]
Agree.
Every decision deserves careful consideration because it will have far-reaching consequences.
Slow down and focus on the important points.

**Comment by andradei** [comment]
I'm more of a lurker and barely used V. It is an interesting project and I watch it with the expectation of one day using it in real projects.

In addition to keeping it consistent and simple like mentioned above, the last thing V needs is a syntax like Rust's that makes one's eye bleed in that see of sigils.

**Comment by medvednikov** [comment]
> Hopefully changes continue to be guided by fundamental core philosophies about the language

@Wajinn they always are. Getting a 51/49 poll result on Twitter would not mean such a fundamental change would instantly happen.

I've had 70/30 polls on Twitter and still kept the design that I originally had in mind.

They are used just to get a public opinion. They are not binding :)
